### ISSUE TITLE
make org.neo4j.test/neo4j-harness a test dependency only (in downstream projects)

### ISSUE THREAD

**Issue by BenSchZA** [gorillalabs/neo4j-clj#19]
Hi there!

I'm encountering an error with `Leiningen 2.9.1 on Java 1.8.0_222 OpenJDK 64-Bit Server VM` and OpenJDK 11 when running the example:

```
Syntax error (ClassNotFoundException) compiling at (neo4j_clj/core.clj:1:1).
org.neo4j.harness.Neo4jBuilders
```

Do you have any idea what would cause this? I'm entirely new to Clojure.

Appreciate any guidance.

**Comment by chrisbetz** [comment]
Hi Ben,

I didn't dig deeper into it, but on my travisCI, I got this error on JDK 10:

```
Syntax error (UnsupportedClassVersionError) compiling at (neo4j_clj/core.clj:1:1).
631org/neo4j/harness/Neo4jBuilders has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 54.0
```

So, I just ran stuff on JDK 11, which is the latest LTS version anyhow. So, if possible switch to JDK11 or up and see if things are running.

Please feel free to re-open this issue if that did not help.

Cheers,

Chris

**Comment by akond** [comment]
I get the same error with `Leiningen 2.9.3 on Java 11.0.6 OpenJDK 64-Bit Server VM`

**Comment by devth** [comment]
Me too.

```
Leiningen 2.9.3 on Java 11.0.6 OpenJDK 64-Bit Server VM
```

**Comment by devth** [comment]
Found [this thread](https://neo4j.com/developer/kb/explanation-of-error-noclassdeffounderror-org-neo4j-kernel-impl-util-jobscheduler/) which explains you need to install the [correct version of APOC](https://github.com/neo4j-contrib/neo4j-apoc-procedures#version-compatibility-matrix).

I was using neo4j Desktop `1.2.4`. When I run `RETURN apoc.version();` it returned `"3.5.0.9"`, so it appears to be an incompatibility issue.

I upgraded to neo4j desktop `1.2.6` (latest as of now), the upgraded my db:

![image](https://user-images.githubusercontent.com/86107/78713419-7d143700-78d7-11ea-8816-ab9872a90604.png)

Next I went to Plugins and insured APOC `4.0.0.6` was installed.

Then I restarted my `lein repl` and verified it worked ⚡️

**Comment by devth** [comment]
Update: it doesn't work after restarting my machine. I'm not sure why yet. Maybe it never worked and I incorrectly verified.

`RETURN apoc.version();` gives `"4.0.0.6"`. Debugging 🤔

**Comment by devth** [comment]
Ok, so it looks like the fix is to add a separate dependency:

```
[org.neo4j.test/neo4j-harness "4.0.0"]
```

`neo4j-clj.core` uses `(org.neo4j.harness Neo4jBuilders Neo4j)` but the project doesn't explicitly depend on `org.neo4j.test/neo4j-harness` (it's only listed as a `provided` dependency).

**Comment by san4d** [comment]
I also ran into this issue:
```bash
$ java -version
openjdk version "11.0.7" 2020-04-14
$ lein -version
Leiningen 2.9.3 on Java 11.0.7 OpenJDK 64-Bit Server VM
```
After adding the `neo4j-harness` dependency, I'm able to compile. However, I'm now facing Issue https://github.com/gorillalabs/neo4j-clj/issues/20.

**Comment by lgessler** [comment]
I also got this error on JDK 11 using deps. Adding the dependency like @devth suggests resolved the issue for me. Would it be appropriate to add `[org.neo4j.test/neo4j-harness "4.0.0"]` to `project.clj`?

**Comment by chrisbetz** [comment]
Hi all,

thanks for being so active and helpful :) I just looked into this and `[org.neo4j.test/neo4j-harness "4.0.0"]` already is in the [project.clj](/project.clj), but not in the [example/project.clj](example/project.clj).

I ran the example from the "wrong" repl, that's why it worked for me.

I altered the [example/project.clj](example/project.clj) to actually contain `[org.neo4j.test/neo4j-harness "4.0.0"]`, so it will be easier for future users.

However, I'll re-open this issue to make sure `[org.neo4j.test/neo4j-harness "4.0.0"]` is only required as a test dependency. Therefore, I'll have to change code.
